File: app/src/main/java/com/olehliskovych/placenote/ui/base/BaseFragment.java
package com.olehliskovych.placenote.ui.base;

import android.os.Bundle;

import dagger.android.support.DaggerFragment;

public abstract class BaseFragment extends DaggerFragment {

    public static BaseFragment newInstance(Class<? extends BaseFragment> fragmentClass) {
        BaseFragment fragment = null;
        try {
            fragment = fragmentClass.newInstance();
        } catch (IllegalAccessException | java.lang.InstantiationException e) {
            e.printStackTrace();
        }

        return fragment;
    }

}

File: app/src/main/java/com/olehliskovych/placenote/ui/base/BaseFragment.kt
package com.olehliskovych.placenote.ui.base

import dagger.android.support.DaggerFragment

abstract class BaseFragment : DaggerFragment() {
    companion object {

        fun newInstance(fragmentClass: Class<out BaseFragment>): BaseFragment? {
            var fragment: BaseFragment? = null
            try {
                fragment = fragmentClass.newInstance()
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
            } catch (e: java.lang.InstantiationException) {
                e.printStackTrace()
            }

            return fragment
        }
    }

}
